python_basics_writing_python_code/python_basics_writing_python_code.md

```diff
@@ -0,0 +1,921 @@
+
+    @sage
+    <div class="python_link">
+    <lia-keep>
+    <script type="text/x-sage">
+
+    print(contact_info.pop("email"))
+print(contact_info)
+
+    </script>
+    </lia-keep>
+    </div>
+
+    <div class="warning">
+    Note that this change is permanent! If you want to keep the original dictionary as-is, you should make a copy of the dictionary by assigning it to a new variable, and then using `.pop()` on the copy. In this example though, all you need to do to restore the dictionary to its original form is re-run the first code cell or refresh the page.
+    </div>
+
+* There isn't a method to add items to a dictionary, but it can be done! You can give the dictionary a new key and assign a value to it (notice that we're using bracket notation again). This is another example of a permanent change to our dictionary.
+
+    @sage
+    <div class="python_link">
+    <lia-keep>
+    <script type="text/x-sage">
+
+    contact_info["work_email"] = "myname@company.com"
+print(contact_info)
+
+    </script>
+    </lia-keep>
+    </div>
+
+  You can also use the same syntax with an exisitng key to assign a new value (note that the new value will **replace** the previous value).
+
+### Quiz: Dictionaries
+
+True or False: Duplicate key-value pairs are allowed in dictionaries.
+
+[( )] True
+[(X)] False
+***
+<div class = "answer">
+While **values** can be repeated in a dictionary, each **key** must be unique. However, you can have multiple values for a key by putting those values in a list.
+</div>
+***
+
+## It's Your Turn!
+
+Ready to practice what you've learned?
+
+A teacher friend of yours can never remember what the grade ranges are for letter grades. They have requested that you write some code that will take a list of final grades and return the letter grades corresponding to the score. After doing some research, you know that:
+
+90-100 = "A"
+
+80-89 = "B"
+
+70-79 = "C"
+
+60-69 = "D"
+
+< 60 = "F"
+
+Your friend also sends you the class's final grades, along with the student identifiers:
+
+st1: 88, st2: 78, st3: 34, st4: 97, st5: 64, st6: 89, st7: 56, st8: 83, st9: 92
+
+Using a for loop and if-else statements, can you write some code that returns a dictionary with the student identifiers as the keys and the letter grades that the students should be assigned as the values? If you get stumped, look at the next page for one possible approach.
+
+**Hint:** To loop through the items (the keys and values) of a dictionary that we'll call `my_dict`, you can use the code `for k,v in my_dict.items()`.
+
+If you do glance at the answer, note that your code might look a little different; as long as you get the output you're looking for, that's okay!
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+#Hint: You'll need the identifiers and the scores in your loop.
+#If you get stuck, try googling something like "Python looping through items in a dictionary".
+
+</script>
+</lia-keep>
+</div>
+
+### Possible solution
```

Comment:
Maybe add an intermediate partial solution so people who struggle can get some help but then also enter some of their own code before seeing a working solution?

Reply:
Ooooh I like that idea!

Reply:
I like it too!